File: super-creator-ultimate/packages/electron/src/main/dataPaths.ts
import fs from 'node:fs';
import path from 'node:path';

const BASE_DIR = process.env.SUPER_CREATOR_DATA_DIR ?? 'E:/SuperCreatorData';

export interface DataPaths {
  baseDir: string;
  browserProfilesDir: string;
  mediaDir: string;
  pythonVenvDir: string;
  whisperModelsDir: string;
  tempDir: string;
  logsDir: string;
  electronUserDataDir: string;
}

export const dataPaths: DataPaths = {
  baseDir: BASE_DIR,
  browserProfilesDir: path.join(BASE_DIR, 'browser-profiles'),
  mediaDir: path.join(BASE_DIR, 'media'),
  pythonVenvDir: path.join(BASE_DIR, 'python-venv'),
  whisperModelsDir: path.join(BASE_DIR, 'whisper-models'),
  tempDir: path.join(BASE_DIR, 'temp'),
  logsDir: path.join(BASE_DIR, 'logs'),
  electronUserDataDir: path.join(BASE_DIR, 'electron-userdata')
};

export const ensureDataDirectories = (): void => {
  Object.values(dataPaths).forEach((dir) => fs.mkdirSync(dir, { recursive: true }));
};

File: super-creator-ultimate/packages/worker/services/data_paths.py
from __future__ import annotations

import os
from pathlib import Path

BASE_DIR = Path(os.environ.get("SUPER_CREATOR_DATA_DIR", "E:/SuperCreatorData"))
MEDIA_DIR = BASE_DIR / "media"
TEMP_DIR = BASE_DIR / "temp"
LOGS_DIR = BASE_DIR / "logs"
WHISPER_MODELS_DIR = Path(os.environ.get("WHISPER_MODEL_DIR", str(BASE_DIR / "whisper-models")))


def ensure_worker_dirs() -> None:
    for directory in (BASE_DIR, MEDIA_DIR, TEMP_DIR, LOGS_DIR, WHISPER_MODELS_DIR):
        directory.mkdir(parents=True, exist_ok=True)


def make_temp_dir(prefix: str) -> Path:
    ensure_worker_dirs()
    target = TEMP_DIR / f"{prefix}-{os.urandom(6).hex()}"
    target.mkdir(parents=True, exist_ok=True)
    return target
